nab/scorer.py

```diff
@@ -18,274 +18,11 @@
-        # thisTP = self.costMatrix["tpWeight"]  # flat scoring
-        scores.iloc[window.indices[0]] = thisTP
-        tpScore += thisTP
-
-    # Go through each false positive and score it. Each FP leads to a negative
-    # contribution dependent on how far it is from the previous window.
-    fpLabels = self.data[self.data["type"] == "fp"]
-    fpScore = 0
-    for i in fpLabels.index:
-      windowId = self.getClosestPrecedingWindow(i)
-
-      if windowId == -1:
-        thisFP = -self.costMatrix["fpWeight"]
-        scores.iloc[i] = thisFP
-        fpScore += thisFP
-      else:
-        window = self.windows[windowId]
-        position = abs(window.indices[-1] - i)/float(window.length-1)
-        thisFP = scaledSigmoid(position)*self.costMatrix["fpWeight"]
-        # thisFP = -self.costMatrix["fpWeight"]  # flat scoring
-        scores.iloc[i] = thisFP
-        fpScore += thisFP
-
-    self.score = tpScore + fpScore + fnScore
-
-    return (scores, self.score)
-
-
-  def getClosestPrecedingWindow(self, index):
-    """Given a record index, find the closest preceding window.
-
-    This helps score false positives.
-
-    @param  index   (int)   Index of a record.
-
-    @return         (int)   Window id for the last window preceding the given
-                            index.
-    """
-    minDistance = float("inf")
-    windowId = -1
-    for window in self.windows:
-      if window.indices[-1] < index:
-        dist = index - window.indices[-1]
-        if dist < minDistance:
-          minDistance = dist
-          windowId = window.id
-
-    return windowId
-
-
-def sigmoid(x):
-  """Standard sigmoid function."""
-  return 1 / (1 + math.exp(-x))
-
-
-def scaledSigmoid(relativePositionInWindow):
-  """Return a scaled sigmoid function given a relative position within a
-  labeled window.  The function is computed as follows:
-
-  A relative position of -1.0 is the far left edge of the anomaly window and
-  corresponds to S = 2*sigmoid(5) - 1.0 = 0.98661.  This is the earliest to be
-  counted as a true positive.
-
-  A relative position of -0.5 is halfway into the anomaly window and
-  corresponds to S = 2*sigmoid(0.5*5) - 1.0 = 0.84828.
-
-  A relative position of 0.0 consists of the right edge of the window and
-  corresponds to S = 2*sigmoid(0) - 1 = 0.0.
-
-  Relative positions > 0 correspond to false positives increasingly far away
-  from the right edge of the window. A relative position of 1.0 is past the
-  right  edge of the  window and corresponds to a score of 2*sigmoid(-5) - 1.0 =
-  -0.98661.
-
-  @param  relativePositionInWindow (float)  A relative position
-                                            within a window calculated per the
-                                            rules above.
-
-  @return (float)
-  """
-  if relativePositionInWindow > 3.0:
-    # FP well behind window
-    return -1.0
-  else:
-    return 2*sigmoid(-5*relativePositionInWindow) - 1.0
```

Comment:
This is the primary chunk of logic being replaced. The rest of the changes in this file are plumbing. All this logic can now be found in `sweeper.py`. (note to self)